configure

```diff
@@ -121,6 +121,17 @@ detect_openmp () {
       echo "no"
     fi
 
+    # https://github.com/Rdatatable/data.table/issues/6409
+    printf "%s" "* checking if R installation supports OpenMP with \"-fopenmp\" ... "
```

Comment:
LGTM! 
To avoid the slight repetition with the check above though, you could consider merging them (if it still maintains readability), maybe something like:
```r
printf "%s" "* Checking if R installation supports OpenMP with \"-Xclang -fopenmp\" or \"-fopenmp\" ... "
if CPPFLAGS="${CPPFLAGS} -Xclang -fopenmp" PKG_LIBS="-lomp" "${R_HOME}/bin/R" CMD SHLIB test-omp.c >> config.log 2>&1 || \
   CPPFLAGS="${CPPFLAGS} -fopenmp" "${R_HOME}/bin/R" CMD SHLIB test-omp.c >> config.log 2>&1; then
  echo "yes"
  if ${CC} --version 2>&1 | grep -q "clang"; then
    PKG_CFLAGS="${PKG_CFLAGS} -Xclang -fopenmp"
    PKG_LIBS="${PKG_LIBS} -lomp"
  else # This probably supports a few other GCC-compatible compilers too I reckon (such as ICC and PGI perhaps), so having a generic else sounds reasonable as opposed to being checked for GCC only.
    PKG_CFLAGS="${PKG_CFLAGS} -fopenmp"
  fi
  export PKG_CFLAGS
  export PKG_LIBS
  export R_OPENMP_ENABLED=1
  return
else
  echo "no"
fi
```
instead of:
```r
printf "%s" "* checking if R installation supports OpenMP with \"-Xclang -fopenmp\" ... "
if CPPFLAGS="${CPPFLAGS} -Xclang -fopenmp" PKG_LIBS="-lomp" "${R_HOME}/bin/R" CMD SHLIB test-omp.c >> config.log 2>&1; then
  echo "yes"
  export PKG_CFLAGS="${PKG_CFLAGS} -Xclang -fopenmp"
  export PKG_LIBS="${PKG_LIBS} -lomp"
  export R_OPENMP_ENABLED=1
  return
else
  echo "no"
fi

printf "%s" "* checking if R installation supports OpenMP with \"-fopenmp\" ... "
if CPPFLAGS="${CPPFLAGS} -fopenmp" "${R_HOME}/bin/R" CMD SHLIB test-omp.c >> config.log 2>&1; then
  echo "yes"
  export PKG_CFLAGS="${PKG_CFLAGS} -fopenmp"
  export R_OPENMP_ENABLED=1
  return
else
  echo "no"
fi
```

Reply:
Unrelated to this PR, but a minor nit while we're discussing this file: openmp > 'OpenMP' here: (for consistency with the other messages)
https://github.com/Rdatatable/data.table/blob/392a321689f32113d58e6f09b4232adbfee05d5c/configure#L99

Reply:
Unrelated to this PR, but a minor nit while we're discussing this file: 'openmp' > 'OpenMP' here: (for consistency with the other messages)
https://github.com/Rdatatable/data.table/blob/392a321689f32113d58e6f09b4232adbfee05d5c/configure#L99

Reply:
Unrelated to this PR, but a minor nit while we're discussing this file: openmp > 'OpenMP' here: (for consistency with the other messages)
https://github.com/Rdatatable/data.table/blob/392a321689f32113d58e6f09b4232adbfee05d5c/configure#L99

Reply:
Also: Unrelated to this PR, but a minor nit while we're discussing this file: 'openmp' > 'OpenMP' here: (for consistency with the other messages)
https://github.com/Rdatatable/data.table/blob/392a321689f32113d58e6f09b4232adbfee05d5c/configure#L99

Reply:
Also: Unrelated to this PR, but a minor nit while we're discussing this file - 'openmp' > 'OpenMP' here: (for consistency with the other messages)
https://github.com/Rdatatable/data.table/blob/392a321689f32113d58e6f09b4232adbfee05d5c/configure#L99